src/sentry/tasks/user_reports.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import
+
+from sentry.models import Group, UserReport
+from sentry.tasks.base import instrumented_task, retry
+
+
+@instrumented_task(
+    name='sentry.tasks.user_reports.backfill_group',
+    queue='cleanup',
+    default_retry_delay=60,
+    max_retries=5,
+)
+@retry(exclude=(UserReport.DoesNotExist, ))
+def backfill_group(report_id, **kwargs):
+    report = UserReport.objects.filter(
+        id=report_id,
+    ).select_related('project').get()
+
+    if report.group_id is None:
+        report.group = Group.objects.from_event_id(
```

Comment:
Don't we want to also check that this isn't just setting it to None again? Since there might be a period of time, and we'd want to trigger a retry later if this was still None.

Reply:
soo i might be confused about how `retry` works, but from reading the code, what i thought would happen here is that `from_event_id` will raise a `Group.DoesNotExist` and then retry will retry again, up to 5 times, after 60 seconds. Is that not right?

Reply:
Oh, you might be right. I'm not 100% sure how celery handles this stuff tbh. We can try it and see what happens though. I also thought, even though I wrote this, that `from_event_id` returned `None`. :confused: